### PR TITLE
docs(cra): trouble shooting cra4

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

image

- Add CRA 4 mjs trouble shooting
- Fix typo on CRA Geo components trouble shooting
- Change page organization
- Add alert saying that only CRA5 is supported

image

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx [hunk 1]
```diff
@@ -189,7 +195,9 @@ export default defineConfig({
 
 ## Create React App
 
-When using [Geo components](/connected-components/geo) and Create React App v5, users may experience the following error when rendering the `<MapView>` component in a production build:
+### CRA 4:`Uncaught ReferenceError: g is not defined`
+
+When using [Geo components](/connected-components/geo) and Create React App v4, users may experience the following error when rendering the `<MapView>` component in a production build:
```
Comment (ioanabrooks) [inline docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx:200]: The original text was referencing CRA 5, is this change intentional?

File: docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx [hunk 2]
```diff
@@ -220,11 +228,39 @@ to the following:
...
+
+Create React App v4 is not officially supported by Amplify UI. When you use it, you may get the following error
+
+```bash
+ ./node_modules/@aws-amplify/ui-react/dist/esm/components/..\/\*.mjs
+
+ Can’t import the named export ‘Amplify’ from non EcmaScript module (only default export is available)
+```
+
+To resolve the error, you may either (1) upgrade to Create React App version 5 ([Migration Guide](https://github.com/facebook/create-react-app/blob/main/CHANGELOG.md#migrating-from-40x-to-500)), or (2) to over-write the webpack config by using tools like [React App Rewired](https://github.com/timarney/react-app-rewired), [Craco](https://craco.js.org/docs/configuration/webpack/), using the following line:
```
Comment (ioanabrooks) [inline docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx:241]: nit: `over-write` -> `overwrite`

File: docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx (hunk 1 above)
Comment (0618) [reply docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx:200]: Good catch. Yes. I was about to comment that. 

It fixed a typo. Should be CRA4 https://github.com/aws-amplify/amplify-ui/issues/1656

File: docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx (hunk 2 above)
Comment (0618) [reply docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx]: Done in f83847aba81caa942990d79ed350690dbf9ee945

File: docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx (hunk 1 above)
Comment (ioanabrooks) [reply docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx:200]: Thanks for making these changes! Love that the table of contents makes all the troubleshooting items visible for better discoverability!

**Review by zchenwei — DISMISSED** [review]
LGTM! Thanks for your effort here!

File: docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx [hunk 3]
```diff
@@ -189,7 +195,9 @@ export default defineConfig({
 
 ## Create React App
 
-When using [Geo components](/connected-components/geo) and Create React App v5, users may experience the following error when rendering the `<MapView>` component in a production build:
+### CRA 4:`Uncaught ReferenceError: g is not defined`
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx:198]: Missing space
```suggestion
### CRA 4: `Uncaught ReferenceError: g is not defined`
```

File: docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx [hunk 4]
```diff
@@ -220,11 +228,39 @@ to the following:
...
+
+Create React App v4 is not officially supported by Amplify UI. When you use it, you may get the following error
+
+```bash
+ ./node_modules/@aws-amplify/ui-react/dist/esm/components/..\/\*.mjs
+
+ Can’t import the named export ‘Amplify’ from non EcmaScript module (only default export is available)
+```
+
+To resolve the error, you may either (1) upgrade to Create React App version 5 ([Migration Guide](https://github.com/facebook/create-react-app/blob/main/CHANGELOG.md#migrating-from-40x-to-500)), or (2) to overwrite the webpack config by using tools like [React App Rewired](https://github.com/timarney/react-app-rewired), [Craco](https://craco.js.org/docs/configuration/webpack/), using the following line:
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx:241]:
```suggestion
To resolve the error, you may either (1) upgrade to Create React App version 5 ([Migration Guide](https://github.com/facebook/create-react-app/blob/main/CHANGELOG.md#migrating-from-40x-to-500)), or (2) override the webpack config using tools like [React App Rewired](https://github.com/timarney/react-app-rewired), [Craco](https://craco.js.org/docs/configuration/webpack/) to add the following rule:
```

File: docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx (hunk 1 above)
Comment (0618) [reply docs/src/pages/[platform]/getting-started/troubleshooting/troubleshooting.react.mdx:200]: Kudos to @zchenwei for telling me that 🙌

**Review by ioanabrooks — APPROVED** [review]

File: docs/src/pages/[platform]/getting-started/usage/create-react-app/react.mdx [hunk 5]
```diff
@@ -1,8 +1,13 @@
-import { Tabs, TabItem } from '@aws-amplify/ui-react';
+import { Alert,Tabs, TabItem } from '@aws-amplify/ui-react';
```
Comment (zchenwei) [inline docs/src/pages/[platform]/getting-started/usage/create-react-app/react.mdx:1]: I found a nit 😈 
```suggestion
import { Alert, Tabs, TabItem } from '@aws-amplify/ui-react';
```

**Review by zchenwei — DISMISSED** [review]

**Review by zchenwei — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**